### PR TITLE
Add NanosecondsSinceEpochInfo

### LINKED ISSUE #751: Implement TimerInfo
This could go into `GAPInfo` or one of its substructures, maybe `KernelInfo`.

### PR DESCRIPTION
**Pull request by markuspf**
Implement `NanosecondsSinceEpochInfo` and document. This fixes #751.

### CODE REVIEW

File: src/gap.c [hunk 1]
```diff
@@ -900,6 +900,40 @@ Obj FuncNanosecondsSinceEpoch(Obj self)
...
+*F  FuncNanosecondsSinceEpochInfo( <self> )
+**
+**  'FuncNanosecondsSinceEpochInformation' returns a plain record
+**  contains information about the timers used for FuncNanosecondsSinceEpoch
+**
+*/
+Obj FuncNanosecondsSinceEpochInfo(Obj self)
+{
+  Obj res, tmp;
+  Int resolution;
```
Comment (ChrisJefferson) [inline src/gap.c]: This should be Int8 (and in the method as well), as the resolution could fill a 32-bit integer I think

Comment (markuspf) [inline src/gap.c]: I'd hope that resolution does not fill a 32bit integer (its supposed to be the smallest possible increment in nanoseconds that one gets back between two calls to `NanosecondsSinceEpoch`), but at least in `timespec` it's potentially a `time_t` and a `long` (with maximum value of 999999999).

I also have to admit I never really looked at the typedefs for the different int types before, I'll change this to Int8.

**Comment by markuspf** [comment]
Incidentally @ChrisJefferson, can you confirm that the method to get the resolution on darwin is correct?

File: src/system.h [hunk 2]
```diff
@@ -1031,7 +1031,11 @@ extern void SyExit (
 **  Returns -1 to represent failure
 **
 */
-Int8 SyNanosecondsSinceEpoch();
+extern Int8 SyNanosecondsSinceEpoch();
+extern Int8 SyNanosecondsSinceEpochResolution();
+
+extern const char *SyNanosecondsSinceEpochMethod;
+extern Int SyNanosecondsSinceEpochMonotonic;
```
Comment (fingolfin) [inline src/system.h]: Perhaps make SyNanosecondsSinceEpochMonotonic `const`? And change `SyNanosecondsSinceEpochMethod` similarly to `const char * const`?

File: src/system.c [hunk 3]
```diff
@@ -1499,6 +1499,77 @@ Int8 SyNanosecondsSinceEpoch()
...
+  const char *SyNanosecondsSinceEpochMethod = "mach_absolute_time";
+  Int SyNanosecondsSinceEpochMonotonic = 1;
+#elif defined(HAVE_CLOCK_GETTIME) && defined(CLOCK_MONOTONIC)
+  const char *SyNanosecondsSinceEpochMethod = "clock_gettime";
+  Int SyNanosecondsSinceEpochMonotonic = 1;
+#elif defined(HAVE_GETTIMEOFDAY)
+  const char *SyNanosecondsSinceEpochMethod = "gettimeofday";
+  Int SyNanosecondsSinceEpochMonotonic = 0;
+#else
+  const char *SyNanosecondsSinceEpochMethod = "";
```
Comment (fingolfin) [inline src/system.c]: How about setting `SyNanosecondsSinceEpochMethod` to e.g. `"none"` or `"unsupported"` in this case? To make it more explicit what's going on.

**Comment by markuspf** [comment]
Agreed and changed.

**Pull request merged**